sync/limiter.go

```diff
@@ -0,0 +1,90 @@
+package sync
+
+import (
+	"sync"
+)
+
+// A Limiter restricts access to a resource.
```

Comment:
I think this docstring could be better. Reading this it's hard to get an idea of the high level goals of the `Limiter`, where a `Limiter` is useful, and what makes it different from other synchronization abstractions that restrict access to resources.